### ISSUE TITLE
Allow COMMA again in NoWhiteSpaceBefore

### ISSUE THREAD

**Issue by wstomv** [checkstyle/checkstyle#2089]
At some point, COMMA was dropped from the tokens to which NoWhiteSpaceBefore can be applied.

I suggest to add it back in, or explain in the documentation why it was removed.

**Comment by romani** [comment]
that was not intentional , we just enforced getAcepptableTokens validaton 
list of acceptable tokens was created at 2002-2003 year - https://github.com/checkstyle/checkstyle/blame/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java#L85

Please provide your configuration for this check and small code chunk(that followed that format) to let validate your intend and extend that list.

**Comment by wstomv** [comment]
My request is to allow COMMA in NoWhitespaceBefore, and to report errors for it.

My 19-month old configuration file contains this check:

        <module name="NoWhitespaceBefore">
            <property name="tokens" value="COMMA, SEMI"/>
        </module>

and is intended to capture such formatting errors as

```java
        f(1 , 2); // space before and after ,
```

where a comma is preceded by whitespace. Checkstyle 5.7 accepted this configuration. However, upon closer inspection, it does not catch this problem with whitespace preceding the comma. But it should have caught it, in my opinion.

With Checkstyle 6.10.1, I get an error message on the configuration file:

    cannot initialize module TreeWalker - Token "COMMA" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceBeforeCheck

So, please, allow COMMA in NoWhitespaceBefore and perform that check.

**Comment by romani** [comment]
merged, added as default token also, enforced over Checkstyle code.
